lib/spack/spack/environment.py

```diff
@@ -2129,6 +2130,17 @@ def is_latest_format(manifest):
     return not changed
 
 
+@contextlib.contextmanager
+def deactivate_environment():
```

Comment:
It'd be better to work with spack.environment.activate() and spack.environment.deactivate() to avoid breaking invariants. That'd be much easier after https://github.com/spack/spack/pull/25564:

```python
e = ev.active_environment()
if not e:
  yield
else:
  ev.deactivate()
  yield
  ev.activate(e)
```

Reply:
I agree, but shell code is not factored out. That's exactly the reason why I didn't use activate / deactivate here.

Reply:
well, then we can make that pr a bit higher priority? :D

Reply:
Let me try to just cherry-pick that commit and submit a separate pr.

Reply:
> well, then we can make that pr a bit higher priority? :D

In general I wouldn't hold up a bugfix on a large refactor PR, but I see how that can work to ask to raise priority :stuck_out_tongue: I'm absolutely fine with a focused cherry-pick. In that case the second PR to go in would refactor things correctly.

Reply:
@haampie I think it's probably easier to get the bugfix in first, then refactor this method as part of the refactor PR, rather than the other way around.